nextcord/automod.py

```diff
@@ -0,0 +1,333 @@
+        self.exempt_channel_ids: List[int] = [
+            int(exempt_channel) for exempt_channel in data["exempt_channels"]
+        ]
+        self.keyword_filters: Optional[List[str]] = None
+        self.presets: Optional[List[KeywordPresetType]] = None
+
+        self._unpack_trigger_metadata(data["trigger_metadata"])
+        self.actions: List[AutoModerationAction] = []
+
+        for action in data["actions"]:
+            self.actions.append(AutoModerationAction(data=action))
+
+    def _unpack_trigger_metadata(self, trigger_metadata: TriggerMetadataPayload):
+        if trigger_metadata.get("keyword_filter") is not None:
+            self.keyword_filters = trigger_metadata["keyword_filter"]  # type: ignore
+        if trigger_metadata.get("presets") is not None:
+            self.presets = [try_enum(KeywordPresetType, preset) for preset in trigger_metadata["presets"]]  # type: ignore -- pylint messed up somehow
+
+    async def delete(self):
+        """Delete this auto moderation rule."""
+        await self._state.http.delete_automod_rule(guild_id=self.guild.id, rule_id=self.id)
+
+    @property
+    def creator(self):
+        """Optional[:class:`Member`]: The member that created this rule."""
+        return self.guild.get_member(self.creator_id)
+
+    @property
+    def exempt_roles(self) -> List[Role]:
+        """List[:class:`Role`]: A list of role that will not be affected by this rule. `[]` if not set."""
+        return [self.guild.get_role(exempt_role_id) for exempt_role_id in self.exempt_role_ids]  # type: ignore -- they can't be None.
+
+    @property
+    def exempt_channels(self) -> List[GuildChannel]:
+        """List[:class:`GuildChannel`]: A list of channels that will not be affected by this rule. `[]` if not set."""
+        return [self.guild.get_channel(exempt_channel_id) for exempt_channel_id in self.exempt_channel_ids]  # type: ignore -- same
+
+    @overload
+    async def edit(
+        self,
+        *,
+        name: Optional[str] = ...,
+        event_type: Optional[EventType] = ...,
+        keyword_filters: Optional[List[str]] = ...,
+        notify_channel: Optional[GuildChannel] = ...,
+        timeout_seconds: Optional[int] = ...,
+        enabled: Optional[bool] = ...,
+        exempt_roles: Optional[List[Role]] = ...,
+        exempt_channels: Optional[List[GuildChannel]] = ...,
+    ):
+        ...
+
+    @overload
+    async def edit(
+        self,
+        *,
+        name: Optional[str] = ...,
+        event_type: Optional[EventType] = ...,
+        preset: Optional[KeywordPresetType] = ...,
+        notify_channel: Optional[GuildChannel] = ...,
+        timeout_seconds: Optional[int] = ...,
+        enabled: Optional[bool] = ...,
+        exempt_roles: Optional[List[Role]] = ...,
+        exempt_channels: Optional[List[GuildChannel]] = ...,
+    ):
+        ...
+
+    async def edit(self, **fields):
```

Comment:
Lets not continue the dpy trend of kwargs. Please use actual arguments

Reply:
As a follow on, this will also make the init nicer. But thats for a future review